Tella/Data/Networking/APICall/URLRequest.swift

```diff
@@ -12,26 +12,49 @@ typealias APIDataResponse = AnyPublisher<(Data,[AnyHashable:Any]?), APIError>
 protocol WebRepository {}
 
 extension WebRepository {
-    func getAPIResponse<Value>(endpoint: any APIRequest) -> APIResponse<Value>
-    where Value: Decodable {
+    private func fetchData(endpoint: any APIRequest) -> AnyPublisher<(data: Data, response: URLResponse), Error> {
         do {
-            guard (NetworkMonitor.shared.isConnected) else {
+            guard NetworkMonitor.shared.isConnected else {
                 return Fail(error: APIError.noInternetConnection)
                     .eraseToAnyPublisher()
             }
             let request = try endpoint.urlRequest()
-            request.curlRepresentation()
             let configuration = URLSessionConfiguration.default
             configuration.waitsForConnectivity = false
-            
+
             return URLSession(configuration: configuration)
                 .dataTaskPublisher(for: request)
-                .requestJSON()
-        } catch _ {
+                .mapError { $0 as Error }
+                .eraseToAnyPublisher()
+        } catch {
             return Fail(error: APIError.invalidURL)
                 .eraseToAnyPublisher()
         }
     }
+
+    func getAPIResponse<Value>(endpoint: any APIRequest) -> APIResponse<Value>
+    where Value: Decodable {
+        fetchData(endpoint: endpoint)
+            .requestJSON()
+            .eraseToAnyPublisher()
+    }
+
+    func getAPIResponseForBinaryData(endpoint: any APIRequest) -> APIResponse<Data> {
+        fetchData(endpoint: endpoint)
+            .tryMap { output in
```

Comment:
Maybe we can use `.requestData() .eraseToAnyPublisher()` instead of  ` .tryMap ... .mapError ...`